### PR TITLE
1822 mx - P16 and start of builder cubes

### PR DESCRIPTION
**Pull request by outsidePasser**

### CODE REVIEW

File: public/icons/1822_mx/red_cube.svg [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (tobymao) [inline public/icons/1822_mx/red_cube.svg:1]: minify this

File: lib/engine/step/special_choose.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      def abilities(entity, **kwargs, &block)
+        @game.abilities(entity, :choose_ability, **kwargs, &block)
+      end
+
+      def description
+        'Choose'
+      end
+
+      def process_choose_ability(action)
+        raise NotImplementedError
```
Comment (tobymao) [inline lib/engine/step/special_choose.rb:35]: what's this

File: assets/app/view/game/abilities.rb [hunk 3]
```diff
@@ -104,13 +104,7 @@ def render_sell_company_button
       end
 
       def render_ability_choice_buttons
-        step = if @game.round.active_step.respond_to?(:choices_ability)
```
Comment (tobymao) [inline assets/app/view/game/abilities.rb:107]: why was this changed

File: lib/engine/ability/choose_ability.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class ChooseAbility < Base
```
Comment (tobymao) [inline lib/engine/ability/choose_ability.rb:7]: why is this needed

**Comment by outsidePasser** [comment]
The problem I thought I had in 1822 turned out to be a false alarm.  These changes are working for 1822.

File: assets/app/view/game/abilities.rb [hunk 5]
```diff
@@ -104,13 +104,7 @@ def render_sell_company_button
...
-                 @game.round.active_step
-               else
-                 @game.round.step_for(
-                  @selected_company, 'choose_ability'
-                )
-               end
+        step = @game.round.step_for(@selected_company, 'choose_ability')
```
Comment (tobymao) [inline assets/app/view/game/abilities.rb:107]: @outsidePasser you didn't answer my question

File: lib/engine/ability/choose_ability.rb (hunk 4 above)
Comment (tobymao) [reply lib/engine/ability/choose_ability.rb:7]: @outsidePasser you didn't answer my question

File: lib/engine/game/g_1822_mx/game.rb [hunk 6]
```diff
@@ -386,6 +385,17 @@ def setup
 
           # Setup the NdeM
           setup_ndem
+
+          #          e14_tile = hexes.find { |h| h.id == 'E14' }.tile
```
Comment (tobymao) [inline lib/engine/game/g_1822_mx/game.rb:389]: delete this

File: public/icons/1822_mx/red_cube.svg [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tobymao) [inline public/icons/1822_mx/red_cube.svg:1]: minify this

File: lib/engine/ability/choose_ability.rb (hunk 4 above)
Comment (crericha) [reply lib/engine/ability/choose_ability.rb:7]: I asked him to create a choose ability class instead of hardcoding into the game class like 1822 did, so the choices can be encapsulated into the company.

File: lib/engine/step/special_choose.rb (hunk 2 above)
Comment (crericha) [reply lib/engine/step/special_choose.rb:35]: A base class to make using the choose ability DRY. Processing the response is specific per implementation.

**Review by tobymao — APPROVED** [review]
